
syntax = "proto3";

package lemming.dataplane.sai;

import "dataplane/standalone/proto/common.proto";

option go_package = "github.com/openconfig/lemming/dataplane/standalone/proto";

enum IsolationGroupAttr {
  ISOLATION_GROUP_ATTR_UNSPECIFIED = 0;
  ISOLATION_GROUP_ATTR_TYPE = 1;
  ISOLATION_GROUP_ATTR_ISOLATION_MEMBER_LIST = 2;
}

enum IsolationGroupMemberAttr {
  ISOLATION_GROUP_MEMBER_ATTR_UNSPECIFIED = 0;
  ISOLATION_GROUP_MEMBER_ATTR_ISOLATION_GROUP_ID = 1;
  ISOLATION_GROUP_MEMBER_ATTR_ISOLATION_OBJECT = 2;
}

message CreateIsolationGroupRequest {
  uint64 switch = 1;
  optional IsolationGroupType type = 2 [(attr_enum_value) = 1];
}

message CreateIsolationGroupResponse {
  uint64 oid = 1;
}

message RemoveIsolationGroupRequest {
  uint64 oid = 1;
}

message RemoveIsolationGroupResponse {}

message GetIsolationGroupAttributeRequest {
  uint64 oid = 1;
  repeated IsolationGroupAttr attr_type = 2;
}

message GetIsolationGroupAttributeResponse {
  IsolationGroupAttribute attr = 1;
}

message CreateIsolationGroupMemberRequest {
  uint64 switch = 1;
  optional uint64 isolation_group_id = 2 [(attr_enum_value) = 1];
  optional uint64 isolation_object = 3 [(attr_enum_value) = 2];
}

message CreateIsolationGroupMemberResponse {
  uint64 oid = 1;
}

message RemoveIsolationGroupMemberRequest {
  uint64 oid = 1;
}

message RemoveIsolationGroupMemberResponse {}

message GetIsolationGroupMemberAttributeRequest {
  uint64 oid = 1;
  repeated IsolationGroupMemberAttr attr_type = 2;
}

message GetIsolationGroupMemberAttributeResponse {
  IsolationGroupMemberAttribute attr = 1;
}

service IsolationGroup {
  rpc CreateIsolationGroup(CreateIsolationGroupRequest)
      returns (CreateIsolationGroupResponse) {}
  rpc RemoveIsolationGroup(RemoveIsolationGroupRequest)
      returns (RemoveIsolationGroupResponse) {}
  rpc GetIsolationGroupAttribute(GetIsolationGroupAttributeRequest)
      returns (GetIsolationGroupAttributeResponse) {}
  rpc CreateIsolationGroupMember(CreateIsolationGroupMemberRequest)
      returns (CreateIsolationGroupMemberResponse) {}
  rpc RemoveIsolationGroupMember(RemoveIsolationGroupMemberRequest)
      returns (RemoveIsolationGroupMemberResponse) {}
  rpc GetIsolationGroupMemberAttribute(GetIsolationGroupMemberAttributeRequest)
      returns (GetIsolationGroupMemberAttributeResponse) {}
}
